### PR TITLE
Migration trial: Import flow: Create migration trial step

### PR DESCRIPTION
**Pull request by bogiii**
Related to https://github.com/Automattic/wp-calypso/issues/79968

## Proposed Changes

* Create **Migration trial** step
* Registered the step inside the import-focused flow

## Testing Instructions

* Go to `/setup/import-focused/migrationTrial?siteSlug={SITE_SLUG}&from={JN_SITE}&option=everything`
* Check if there is migrationTrial step there
* Click on `every feature in the Business plan` link and check if there is a modal with a feature list rendered
* Click `Continue` or close the modal
* Click back and check if redirect to the prev "Upgrade Plan" step

**NOTE:** The handler of "Start the trial and migrate" CTA will be implemented in the next iterations

Screen Capture on 2023-08-01 at 12-11-31

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	site: SiteDetails;
+}
+const TrialPlan = function TrialPlan( props: Props ) {
+	const { __ } = useI18n();
+	const { site } = props;
+	const [ showPlanFeaturesModal, setShowPlanFeaturesModal ] = useState( false );
+
+	const targetSiteEligibleForProPlan = useSelector( ( state ) =>
+		isEligibleForProPlan( state, site?.ID )
+	);
```
Comment (p-jackson) [inline client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx]: It's up to you, but I don't think we need to include any of this Pro plan logic. There's still a feature flag for it, but we've basically abandoned the Pro plan for new sites as far as I can tell.

File: client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+							{ planName: plan?.getTitle() }
+						),
+						{
+							a: createElement( 'a', {
+								href: localizeUrl( 'https://wordpress.com/pricing/' ),
+								onClick: ( e: React.MouseEvent< HTMLElement > ) => {
+									e.preventDefault();
+									setShowPlanFeaturesModal( true );
+								},
+							} ),
```
Comment (p-jackson) [inline client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx]: You can also use JSX syntax to create this anchor element.
It does the same thing, but the advantage is the onClick handler's type is inferred 🙂

```suggestion
							a: (
								<a
									href={ localizeUrl( 'https://wordpress.com/pricing/' ) }
									onClick={ ( e ) => {
										e.preventDefault();
										setShowPlanFeaturesModal( true );
									} }
								/>
							),
```

**Review by p-jackson — APPROVED** [review]
The new page looks good to me.
Looking forward to the "Next" button logic being connected 😊

File: client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx (hunk 1 above)
Comment (bogiii) [reply client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx]: Thanks for the feedback. The PRO plan conditional logic has been removed!

File: client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx (hunk 2 above)
Comment (bogiii) [reply client/landing/stepper/declarative-flow/internals/steps-repository/migration-trial/trial-plan.tsx]: That makes sense; thanks for the feedback!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8679267

Thank you @bogiii for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8679267).

**Pull request merged**